webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreRepository.scala

```diff
@@ -180,11 +184,13 @@ class ScoreRepositoryImpl(doobie: Doobie) extends ScoreRepository {
 
   }
 
-  override def deleteScore(nodeId: NodeId, scoreId: Option[String]): IOResult[Unit] = {
-    val whereNode = Some(fr"nodeId = ${nodeId.value}")
+  override def deleteScore(nodeId: Seq[NodeId], scoreId: Option[String]): IOResult[Unit] = {
+    val whereNode = Some(fr"nodeId = ?")
     val whereName = scoreId.map(n => fr"scoreId = ${n}")
     val where     = Fragments.whereAndOpt(whereNode, whereName)
-    val q         = sql"delete from scoredetails " ++ where
-    transactIOResult(s"error when deleting score for node ${nodeId.value}")(xa => q.update.run.transact(xa).unit)
+    val q         = Update[NodeId]((sql"delete from scoredetails " ++ where).toString)
```

Comment:
the `toString` is strange

Reply:
So, `toString` does not give you the sql, but the string `Fragment(the sql here)`

Reply:
You need ```    val whereNode = Some(fr"nodeId = ?")
    val whereName = scoreId.map(n => Fragment.const0(s"scoreId = '${n}'")) // const0 since we don't want it to become '?'
    val where     = Fragments.whereAndOpt(whereNode, whereName)
    val sql       = (sql"delete from scoredetails " ++ where).update.sql
    val q         = Update[NodeId](sql)
    transactIOResult(s"error when deleting score for nodes ${nodeId.map(_.value).mkString(",")}")(xa =>
      q.updateMany(nodeId).transact(xa).unit
    )
```